metadata/src/test/java/org/apache/kafka/controller/errors/EventHandlerExceptionInfoTest.java

```diff
@@ -174,10 +180,18 @@ public void testUnexpectedBaseOffsetExceptionInfo() {
     public void testUnexpectedBaseOffsetFailureMessage() {
         assertEquals("event failed with UnexpectedBaseOffsetException (treated as " +
             "NotControllerException) at epoch 123 in 90 microseconds. Renouncing leadership " +
-            "and reverting to the last committed offset 456.",
+            "and reverting to the last committed offset 456. Detailed exception message: Wanted base offset 3, but the next offset was 4",
                 UNEXPECTED_END_OFFSET.failureMessage(123, OptionalLong.of(90L), true, 456L));
     }
 
+    @Test
+    public void testKafkaExceptionFailureMessage() {
```

Comment:
Should we make it clear we're testing the "isFault" case here in the test name? Perhaps `testFaultExceptionFailureMessage`?

Reply:
It definitely would be better to have a clear name for the test. I've changed it to the suggested one